pool/paymentmgr.go

```diff
@@ -277,33 +289,142 @@ func (pm *PaymentMgr) loadLastPaymentCreatedOn(tx *bolt.Tx) error {
 	return nil
 }
 
-// PPLNSSharePercentages calculates the current mining reward percentages
+// sharePercentages calculates the percentages due each participating account
+// according to their weighted shares.
+func (pm *PaymentMgr) sharePercentages(shares []*Share) (map[string]*big.Rat, error) {
+	totalShares := new(big.Rat)
+	tally := make(map[string]*big.Rat)
+	percentages := make(map[string]*big.Rat)
+
+	// Tally all share weights for each participating account.
+	for _, share := range shares {
+		totalShares = totalShares.Add(totalShares, share.Weight)
+		if _, ok := tally[share.Account]; ok {
+			tally[share.Account] = tally[share.Account].
+				Add(tally[share.Account], share.Weight)
+			continue
+		}
+		tally[share.Account] = share.Weight
+	}
+
+	// Calculate each participating account percentage to be claimed.
+	for account, shareCount := range tally {
+		if tally[account].Cmp(ZeroRat) == 0 {
+			return nil, MakeError(ErrDivideByZero, "division by zero", nil)
+		}
+		accPercent := new(big.Rat).Quo(shareCount, totalShares)
+		percentages[account] = accPercent
+	}
+	return percentages, nil
+}
+
+// PPSEligibleShares fetches all shares within the provided inclusive bounds.
```

Comment:
All shares are returned if `min` is `nil`. We should either mention this in the comment, or create a new func which is specifically for returning all shares and takes no params. 

Personally I think the code would be cleaner with the second option.

Reply:
I think the first option is better here considering the call site, don't see how the code will be cleaner since it'd be calling `PPSEligibleShares` internally and passing nil to it.